File: Classes/Controllers/PBGitWindowController+ToolbarActions.m
#import "PBGitWindowController.h"

#import "GitX-Swift.h"
#import "PBGitRef.h"
#import "PBGitRepository.h"
#import "PBGitRevSpecifier.h"

#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wobjc-protocol-method-implementation"
@implementation PBGitWindowController (PBToolbarActions)

- (IBAction)toolbarFetch:(id)sender
{
	[self fetchAllRemotes:sender];
}

- (IBAction)toolbarPull:(id)sender
{
	PBGitRef *head = self.repository.headRef.ref;
	if (head.isBranch) [self performPullForBranch:head remote:nil rebase:NO];
}

- (IBAction)toolbarPush:(id)sender
{
	PBGitRef *head = self.repository.headRef.ref;
	if (head.isBranch) [self performPushForBranch:head toRemote:nil];
}

- (IBAction)viewRemote:(id)sender
{
	[[PBRepositoryRemoteURLCoordinator shared] viewRemoteForRepository:self.repository presentingWindow:self.window];
}

@end
#pragma clang diagnostic pop

File: Classes/Controllers/PBGitWindowController+ToolbarActions.swift
import AppKit

extension PBGitWindowController {
    @IBAction dynamic func toolbarFetch(_ sender: Any?) {
        fetchAllRemotes(sender)
    }

    @IBAction dynamic func toolbarPull(_ sender: Any?) {
        guard let head = repository?.headRef()?.ref(), head.isBranch else { return }
        performPull(forBranch: head, remote: nil, rebase: false)
    }

    @IBAction dynamic func toolbarPush(_ sender: Any?) {
        guard let head = repository?.headRef()?.ref(), head.isBranch else { return }
        performPush(forBranch: head, toRemote: nil)
    }

    @IBAction dynamic func viewRemote(_ sender: Any?) {
        guard let repository else { return }
        RepositoryRemoteURLCoordinator.shared.viewRemote(repository: repository, presenting: window)
    }
}
